packages/esm-service-queues-app/src/create-queue-entry/queue-fields/queue-fields.component.tsx

```diff
@@ -165,32 +193,54 @@ const QueueFields = React.memo(({ setOnSubmit, defaultInitialServiceQueue }: Que
   }, [onSubmit, setOnSubmit]);
 
   useEffect(() => {
-    if (currentServiceQueueUuid) {
-      setValue('queueService', currentServiceQueueUuid, { shouldValidate: true });
+    if (currentServiceQueueUuid && currentServiceQueueUuid !== queueService) {
+      setValue('queueService', currentServiceQueueUuid, { shouldValidate: isDataLoaded });
     }
-  }, [currentServiceQueueUuid, setValue]);
+  }, [currentServiceQueueUuid, queueService, setValue, isDataLoaded]);
 
   useEffect(() => {
-    if (defaultInitialServiceQueue) {
+    if (defaultInitialServiceQueue && memoizedQueues.length > 0 && !queueService && queueLocation) {
       const initialServiceQueue = memoizedQueues.find((queue) => queue.name === defaultInitialServiceQueue);
-      setValue('queueService', initialServiceQueue?.uuid, { shouldValidate: true });
+      if (initialServiceQueue && queueService !== initialServiceQueue.uuid) {
+        setValue('queueService', initialServiceQueue.uuid, { shouldValidate: isDataLoaded });
+      }
+    }
+  }, [defaultInitialServiceQueue, memoizedQueues, queueService, setValue, isDataLoaded, queueLocation]);
+
+  useEffect(() => {
+    if (queueLocation && queueService) {
+      const isServiceValid = memoizedQueues.some((queue) => queue.uuid === queueService);
+      if (!isServiceValid) {
+        setValue('queueService', '', { shouldValidate: isDataLoaded });
+        setValue('priority', '', { shouldValidate: isDataLoaded });
+      }
+    }
+  }, [queueLocation, memoizedQueues, queueService, setValue, isDataLoaded]);
+
+  useEffect(() => {
+    if (queueService && priorities.length > 0) {
+      const isPriorityValid = priorities.some((p) => p.uuid === priority);
+      if (!isPriorityValid) {
+        const defaultPriority = priorities.find((p) => p.uuid === defaultPriorityConceptUuid) || priorities[0];
+        setValue('priority', defaultPriority.uuid, { shouldValidate: false });
+      }
+    } else if (queueService && priorities.length === 0 && priority !== '') {
+      setValue('priority', '', { shouldValidate: false });
+    } else if (!queueService && priority !== '') {
+      setValue('priority', '', { shouldValidate: false });
+      clearErrors('priority');
     }
-  }, [defaultInitialServiceQueue, memoizedQueues, setValue]);
+  }, [queueService, priorities, priority, defaultPriorityConceptUuid, setValue, clearErrors]);
 
   useEffect(() => {
-    if (memoizedQueueLocations.map((location) => location.id).includes(sessionLocation.uuid)) {
```

Comment:
The PR description mentions:

> Queue location initialization: Form now properly initializes with session location or first available location instead of failing with empty values

Yet this bit of code removes the session prefill. When testing on dev3, the queue location field starts empty and requires the user to explicitly select a queue location. Do we want to restore this behaviour?